rest/src/db/CatapultDb.js

```diff
@@ -401,20 +401,29 @@ class CatapultDb {
 		const sortConditions = { [sortingOptions[options.sortField]]: options.sortDirection };
 		const conditions = await buildConditions();
 
-		return this.queryPagedDocuments(conditions, removedFields, sortConditions, TransactionGroup[group], options);
+		return this.queryPagedDocuments(conditions, removedFields, sortConditions, TransactionGroup[group], options).then(async page => {
+			const data = await this.addBlockMetaToTransactionList(page.data);
+			return ({
+				...page,
+				data
+			});
+		});
 	}
 
 	transactionsByIdsImpl(collectionName, conditions) {
 		return this.queryDocumentsAndCopyIds(collectionName, conditions, { projection: { 'meta.addresses': 0 } })
-			.then(documents => Promise.all(documents.map(document => {
+			.then(list => this.addBlockMetaToTransactionList(list)).then(documents => Promise.all(documents.map(document => {
 				if (!document || !isAggregateType(document))
 					return document;
 
 				return this.queryDependentDocuments(collectionName, [document.id]).then(dependentDocuments => {
 					dependentDocuments.forEach(dependentDocument => {
 						if (!document.transaction.transactions)
 							document.transaction.transactions = [];
-
+						if (document.meta.timestamp && document.meta.feeMultiplier) {
+							dependentDocument.meta.timestamp = document.meta.timestamp;
+							dependentDocument.meta.feeMultiplier = document.meta.feeMultiplier;
+						}
```

Comment:
this does not look right since 0 is valid value for both?
also, is there any way only one of these could be set?

Reply:
Good catch, the timestamp would pass Long(0) but I think feeMultipler is a regular number. I'll update the check for undefined checking